### ISSUE TITLE
Regex issue

### ISSUE THREAD

**Issue by misozask** [silverwind/dnsz#2]
Following regex:
https://github.com/silverwind/dnsz/blob/da99929f0ba2b4dfd20ea5290894dff04aaf0f56/index.js#L20

does not support parsing domains with 'dash', e.g. exa-mple.com
Should be updated to:
```
const re = /^([a-z0-9_.\-@]+)?[\s]*([0-9]+[smhdw]?)?[\s]*([a-z]+)[\s]+([a-z]+)[\s]+(.+)?$/i
```

**Comment by misozask** [comment]
To support also: *.exa-mple.com:
```
const re = /^([a-z0-9_\.\-@\*]+)?[\s]*([0-9]+[smhdw]?)?[\s]*([a-z]+)[\s]+([a-z]+)[\s]+(.+)?$/i
```

**Comment by misozask** [comment]
@silverwind Any chance that you will fix this issue?

**Comment by silverwind** [comment]
Looking into it.

**Comment by silverwind** [comment]
Fixed in `dnsz@1.0.7`.

**Comment by misozask** [comment]
Thanks! :+1:
